### PR TITLE
sys/arduino: Added millis()

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Added arduino-compatilbe `unsigned long millis()`

### Testing procedure

Apply the following patch

```patch
diff --git a/examples/arduino_hello-world/hello-world.sketch b/examples/arduino_hello-world/hello-world.sketch
index bdf414c0fd..4766fbb9e2 100644
--- a/examples/arduino_hello-world/hello-world.sketch
+++ b/examples/arduino_hello-world/hello-world.sketch
@@ -62,6 +62,8 @@ void loop(void)
             Serial.write('\n');
             // reset the count variable
             count = 0;
+            Serial.write("millis(): ");
+            Serial.println((int)millis());
         }
         // else we just remember the incoming char
         else {
```

And flash and run `examples/arduino_hello-world` on your favorite Arduino compatible board. When you type ´<FOO>\n` in the shell, in addition to the `ECHO: <FOO>` you should get the time since the boot of the board in milli seconds.

### Issues/PRs references

None

### CODE REVIEW

File: sys/arduino/base.cpp [hunk 1]
```diff
@@ -72,6 +72,11 @@ unsigned long micros()
     return xtimer_now_usec();
 }
 
+unsigned long millis()
+{
+    return xtimer_now_usec() / 1000;
```
Comment (aabadie) [inline sys/arduino/base.cpp]: You could use `US_PER_MS` instead of hardcoded value 1000. I think @smlng will agree :)

Comment (maribu) [reply sys/arduino/base.cpp]: With the PR being small, I decided to squash right away. And I added a second commit to also update delay to use `US_PER_MS` as well.

File: sys/arduino/base.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ int digitalRead(int pin)
 
 void delay(unsigned long msec)
 {
-    xtimer_usleep(1000 * msec);
+    xtimer_usleep(US_PER_MS * msec);
```
Comment (smlng) [inline sys/arduino/base.cpp]: nit-picky: would be more consistent with other usages if you swap the order, i.e. `msec * US_PER_MS`

other then that, this looks good to go

Comment (maribu) [reply sys/arduino/base.cpp]: done and squashed

**Comment by aabadie** [comment]
Before merging, see https://github.com/RIOT-OS/RIOT/pull/12117#issuecomment-526153367

**Comment by smlng** [comment]
tested with Arduino-uno, works with the patch above. But casting to `int` results in negative values:

```
2019-08-29 14:26:19,355 - INFO # millis(): -24169
```

as @aabadie pointed out: might be nice to have a more advanced example for the Arduino-api but maybe a test would be even better? Anyhow this could go into a separate PR instead.

**Comment by maribu** [comment]
I'll update the test one `Serial.print(unsigned long)` is merged

**Comment by smlng** [comment]
@maribu: from my side this is good to go, even w/o the test - the functionality added here is sound just the print in your example does not work. 

@aabadie any comments? I would ACK and merge this, @maribu why wait?

**Comment by maribu** [comment]
@smlng: Sorry, for the late reply. I assumed that the `print()` / `println()` PR would be small and get merged soon, so that I could the new `Serial.print(unsigned long val)` to properly print the result of `millis()` and extend the Arduino test in `tests`. But things got a bit out of hand and the PR is now a bit more complex.

From my side I have nothing against merging this PR as is. I will add a separate small PR to extend the test later on in that case. But waiting for the other PR and add the test to this PR prior merging would be equally fine to me.

**Comment by keestux** [comment]
Notice that `millis()` will wrap around after 4294967 milliseconds.

**Comment by maribu** [comment]
> Notice that `millis()` will wrap around after 4294967 milliseconds.

Indeed. I think I should have better used `return xtimer_now_usec64() / US_PER_MS;` over `return xtimer_now_usec() / US_PER_MS;`. (On the AVR that would come with a significant cost, but wrapping around 3 orders of magnitude sooner than one would expect might lead to bugs. Especially as we try to mimic the behavior of the Arduino API.)

@keestux: Would you mind to fix this? Otherwise I'll put it on my todo and fix it after my vacation.

**Comment by keestux** [comment]
@maribu I'll create a PR

**Pull request merged**